### PR TITLE
Qlik Replicate Support

### PR DESCRIPTION
**Pull request by rozza**
Qlik Replicate support.

Changes from #18 - squashed and merged initial commit.

 * Renamed Attunity to Qlik
 * Moved towards the MongoDB CDC organization. 
 * Added Qlik specific OperationType enum.
 * Added support for flattened headers eg `{message: {headers_operation: 'INSERT'}, ...}`
 * Simplified logic determining the CDC operation in `OperationHelper`
 * Refactored tests

### CODE REVIEW

**Comment by rozza** [comment]
HI @abraham-leal, this is the cleaned up and refactored version of the qlik support for the 1.5.0 release.  

It would great if you could review.

File: src/test/java/com/mongodb/kafka/connect/sink/cdc/qlik/rdbms/RdbmsHandlerTest.java [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+import com.mongodb.client.model.DeleteOneModel;
+import com.mongodb.client.model.ReplaceOneModel;
+import com.mongodb.client.model.UpdateOneModel;
+import com.mongodb.client.model.WriteModel;
+
+import com.mongodb.kafka.connect.sink.cdc.qlik.OperationType;
+import com.mongodb.kafka.connect.sink.converter.SinkDocument;
+
+@RunWith(JUnitPlatform.class)
+class RdbmsHandlerTest {
```
Comment (abraham-leal) [inline src/test/java/com/mongodb/kafka/connect/sink/cdc/qlik/rdbms/RdbmsHandlerTest.java:50]: I see you handle the Qlik envelope in the code, but there are no tests against the case of no qlik envelope.

**Review by abraham-leal — APPROVED** [review]
LGTM! Just one comment. Thanks @rozza.

File: src/test/java/com/mongodb/kafka/connect/sink/cdc/qlik/rdbms/RdbmsHandlerTest.java (hunk 1 above)
Comment (rozza) [reply src/test/java/com/mongodb/kafka/connect/sink/cdc/qlik/rdbms/RdbmsHandlerTest.java:50]: Hi @abraham-leal 

I think all operation parsing should be covered by the OperationHandlerTest. This test just covers a single type of message so not to duplicate the testing.

If I'm not testing something I should could you provide the json of a sample record so I can investigate?

**Comment by rozza** [comment]
Note: Merged on the command line. 

Waiting to see if extra test scenarios are required.

File: src/test/java/com/mongodb/kafka/connect/sink/cdc/qlik/rdbms/RdbmsHandlerTest.java (hunk 1 above)
Comment (abraham-leal) [reply src/test/java/com/mongodb/kafka/connect/sink/cdc/qlik/rdbms/RdbmsHandlerTest.java:50]: I see only tests for the object created from the OP in `OperationTypeTest`, what I mean is testing for the parsing itself. There can be two types of records by Qlik: With and Without envelope.

With envelope: "{message: { data: {id: 1234, foo: 'bar'}, headers: { operation: 'INSERT'}}}"
Without envelope: "{ data: {id: 1234, foo: 'bar'}, headers: { operation: 'READ'}}"

I see `getSubDocumentOrOriginal` can handle this case, I just don't see it being tested either in `OperationHelper` or `RdbmsHandler`.

Comment (rozza) [reply src/test/java/com/mongodb/kafka/connect/sink/cdc/qlik/rdbms/RdbmsHandlerTest.java:50]: Perfect - we ensure I add tests for that scenario.

Thanks again @abraham-leal

**Pull request closed without merging**